### PR TITLE
 Use weak comparison for If-None-Match header

### PR DESCRIPTION
**Pull request by jbagga**
See https://tools.ietf.org/html/rfc7232#section-3.2 and https://github.com/aspnet/Mvc/issues/7208#issuecomment-357092246

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/FileResultExecutorBase.cs [hunk 1]
```diff
@@ -255,7 +258,7 @@ private static void SetAcceptRangeHeader(HttpResponse response)
                 var state = matchNotFoundState;
                 foreach (var entityTag in etagHeader)
                 {
-                    if (entityTag.Equals(EntityTagHeaderValue.Any) || entityTag.Compare(etag, useStrongComparison: true))
+                    if (entityTag.Equals(EntityTagHeaderValue.Any) || entityTag.Compare(etag, useStrongComparison: useStrongComparison))
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/FileResultExecutorBase.cs]: nit: you can drop the named parameter here, it's already clear what you're doing.

**Comment by rynowak** [comment]
The changes here look fine to me but @Tratcher should give the ultimate signoff. 

Do the tests already cover the important cases? I expected to see a little more churn or more tests added.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/FileResultTest.cs [hunk 2]
```diff
@@ -272,7 +272,7 @@ public void GetPreconditionState_ShouldProcess(string ifMatch, string ifNoneMatc
 
             httpRequestHeaders.IfNoneMatch = ifNoneMatch == null ? null : new[]
             {
-                new EntityTagHeaderValue(ifNoneMatch),
+                new EntityTagHeaderValue(ifNoneMatch, true),
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/FileResultTest.cs]: these could use named params.

**Comment by jbagga** [comment]
@rynowak  @Tratcher The tests we have cover If-None-Match (constructed weakly adding the "W/" prefix [here](https://github.com/aspnet/Mvc/pull/7237/files#diff-b36bb8fbcd4ac1be4c96dc9157e63125R275)) header's  weak comparison to the Etag leading to different preconditions like ShouldProcess, NotModified and PreconditionFailed. I think more granular tests should be covered by EntityTagHeaderValue.Compare tests. If you have more suggestions for missing scenarios let me know!

**Comment by jbagga** [comment]
@Tratcher Updated with minor feedback. What test cases do you think we are missing?

**Comment by Tratcher** [comment]
I was expecting tests with strong and weak tags for If-None-Match

**Comment by jbagga** [comment]
@Tratcher Updated with strong If-None-Match tag tests

**Pull request merged**